abigen/src/contract.rs

```diff
@@ -21,10 +21,24 @@ pub struct Contract {
 
 impl<'a> From<&'a ethabi::Contract> for Contract {
     fn from(c: &'a ethabi::Contract) -> Self {
+        let events: Vec<_> = c
+            .events
+            .values()
+            .flat_map(|events| events.iter().enumerate())
+            .map(|(index, event)| {
+                let name = match index {
+                    0 => event.name.clone(),
+                    _ => format!("{}{}", event.name, index),
```

Comment:
Should we even start the first duplicated one with `1`? My feeling telling me it would be seen right away that there is a duplicate of events, it's more obvious thinking there is more overload when you see `Transfer1`. What do you think?

Reply:
sounds good